features/minion/heartbeat/consumer/src/main/java/org/opennms/minion/heartbeat/consumer/HeartbeatConsumer.java

```diff
@@ -96,6 +103,15 @@
     @Qualifier("eventSubscriptionService")
     private EventSubscriptionService eventSubscriptionService;
 
+    @Autowired
+    private NodeDao nodeDao;
+
+    private final ThreadFactory threadFactory = new ThreadFactoryBuilder()
+            .setNameFormat("minion-provision-handler")
+            .build();
+
+    private final ExecutorService executor = Executors.newSingleThreadExecutor(threadFactory);
```

Comment:
This executor uses an unbound queue by default, so in the event that we were receiving a large number of heartbeat messages and this thread could not keep up, we could still end up in a OOM scenario.

We could mitigate this by limiting the queue capacity to some reasonable number, which would put back-pressure on the broker when full.

Reply:
I think I have a cleaner pattern to deal with this now... let me give it a whirl.